reciperadar/models/recipes/recipe.py

```diff
@@ -55,15 +55,16 @@ def url(self):
     @property
     def products(self):
         unique_products = {
-            ingredient.product.singular: ingredient.product
+            ingredient.product.product.singular: ingredient.product.product
```

Comment:
TODO: cleanup these repetitively-named `product.product` access paths.  The `IngredientProduct` record type and associated table might not be required any more.

Reply:
(best handled in a follow-up changeset so as not to complicate this one)